app/options.go

```diff
@@ -107,6 +107,9 @@ func ServerConnector(s *Server) AppOption {
 		a.searchEngine = s.SearchEngine
 		a.messageExport = s.MessageExport
 		a.metrics = s.Metrics
+		a.ldap = s.Ldap
```

Comment:
Actually, that is not going to work, the `Server.Ldap`, `Server.Saml`, etcetera are late initialized during the first `App.InitServer` execution. the lines before were needed because you need to set up the `App` layer pointers to that variables. Probably the solution is to not use any `App.{ldap,saml,notification,accountMigration}` call, and use the `App.Srv().{Ldap,Saml,Notification,AccountMigration}`.

Anyway, I would revert this part of the changes, and keep the others.

Reply:
Oh I see the problem now. The late initialization isn't the issue. The issue is the the `sync.Once` is done at a server level than at app level. So if you initialize a new app from an existing server, the fields don't get set. Yeah, I think we should remove these app fields and directly use the server fields.

I will send another PR to fix those. Because without fixing the race, this PR isn't much useful.

Reply:
@jespino - This is resolved now after my other PR is now merged to master, and rebased with this. PTAL.